src/_pytest/fixtures.py

```diff
@@ -607,14 +609,11 @@ def _get_active_fixturedef(
     def _get_fixturestack(self) -> List["FixtureDef[Any]"]:
         current = self
         values: List[FixtureDef[Any]] = []
-        while 1:
-            fixturedef = getattr(current, "_fixturedef", None)
-            if fixturedef is None:
-                values.reverse()
-                return values
-            values.append(fixturedef)
-            assert isinstance(current, SubRequest)
+        while isinstance(current, SubRequest):
+            values.append(current._fixturedef)  # type: ignore[has-type]
             current = current._parent_request
+        values.reverse()
```

Comment:
```suggestion
        assert current is None
        values.reverse()
```

Ensures that we don't get weird behaviour if a non-`SubRequest` ever makes it onto the fixture stack.

Reply:
I haven't fully figured out what the `SubRequest` stuff does exactly (quite a spaghetti code there) but your proposed assert doesn't work -- the `_parent_request` eventually reaches a non-`SubRequest` `FixtureRequest` (which `SubRequest` is a subclass of...), not `None`.